syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package NetLib;

// минимальная транспортная информация
message TransportInfo
{
    // уникальный идентификатор отправителя
    optional string sender = 1;
    // список узлов получателей сообщения
    repeated string dest = 2;
};

// Контроль соединения, одна строка таблицы маршрутизации
message RouteLine
{
    // имя узла
    optional string node_name = 1;
    // маршрут до узла
    repeated string routeToClient = 2;
};

// Инициализация регистрация на узле
message RegistrationRequest
{
    // имя регистрирующегося узла
    optional string node_name = 1;
    // пароль
    optional string password = 2;
    // маршруты
    repeated RouteLine routes = 3;
};

// Ответ с результатом регистрации на узле
message RegistrationResponse
{
    // Признак успешной регистрации
    optional uint32 error_code = 1;
    // Имя узла, на котором прошла регистрация в случае успеха
    optional string node_name = 2;

    // маршруты
    repeated RouteLine routes = 3;
};

// Туннельное сообщение
message TunnelData
{
    // Транспортная информация
    optional TransportInfo transport_info = 1;
    // Имя узла, на котором прошла регистрация в случае успеха
    optional bytes tunnelData = 2;
};
